pkg/client/results/junit.go

```diff
@@ -18,25 +18,104 @@ package results
 
 import (
 	"encoding/xml"
+	"fmt"
+	"io"
 	"os"
 	"path/filepath"
 
-	"github.com/onsi/ginkgo/reporters"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
 const (
-	// JunitStdoutKey is the key in the Items.Details map for the system-out output.
-	JunitStdoutKey = "system-out"
+	// JUnitStdoutKey is the key in the Items.Details map for the system-out output.
+	JUnitStdoutKey = "system-out"
 
-	// JunitFailureKey is the key in the Items.Details map for the failure output.
-	JunitFailureKey = "failure"
+	// JUnitFailureKey is the key in the Items.Details map for the failure output.
+	JUnitFailureKey = "failure"
 )
 
-// Filter keeps only the tests that match the predicate function.
-func Filter(predicate func(testCase reporters.JUnitTestCase) bool, testSuite reporters.JUnitTestSuite) []reporters.JUnitTestCase {
-	out := make([]reporters.JUnitTestCase, 0)
+// junitResult is a wrapper around the suite[s] which enable results to
+// be either a single suite or a collection of suites. For instance,
+// e2e tests (which use the onsi/ginkgo reporter) report a single, top-level
+// testsuite whereas other tools report a top-level testsuites object
+// which may have 1+ testsuite children. Only one of the fields should be
+// set, not both.
+type junitResult struct {
```

Comment:
This type is "where the magic happens" since it has a custom unmarshal method which will handle either case based on the top level object.